Update Platform API docs

alecandido opened the pull request.

alecandido committed:
build: Drop pdbpp in favor of pudb

alecandido committed:
docs: Remove legacy fixes for solved sphinx issues

alecandido committed:
docs: Begin platform main description rewriting

alecandido committed:
docs: Describe parameters role and structure

alecandido committed:
docs: Provide detailed description of qubit's role and structure

alecandido committed:
docs: Add alert regarding compilers stability and publicity

alecandido committed:
docs: Describe platform's definition logic

alecandido committed:
docs: Make all code references local

Stripping explicit namespace

alecandido committed:
docs: Complete platform's main documentation

alecandido committed:
docs: Remove old channels section

stavros11 commented on doc/source/tutorials/instrument.rst:
```diff
@@ -131,3 +131,20 @@ Let's see a minimal example:
             return results
+
+
+Dummy platform
+^^^^^^^^^^^^^^
+
+In addition to the real instruments presented above, Qibolab provides the :class:`qibolab.instruments.DummyInstrument`.
+This instrument represents a controller that returns random numbers of the proper shape when executing any pulse sequence.
+This instrument is also part of the dummy platform which is defined in :py:mod:`qibolab._core.dummy` and can be initialized as
+
+.. testcode::  python
+
+    from qibolab import create_platform
+
+    platform = create_platform("dummy")
+
+This platform is equivalent to real platforms in terms of attributes and functions, but returns just random numbers.
+It is useful for testing parts of the code that do not necessarily require access to an actual quantum hardware platform.
```
This paragraph seems to me to fit better in the `main-documention/drivers.rst` where we mainly talk about existing instrument code and `DummyInstrument` is one of these. In this tutorial we are mainly talking about adding new instruments, so dummy doesn't seem much related but I am fine with it in any place.

andrea-pasquale replied:
This is fine for me. In Qibocal we already switch to our own `dummy` called `mock` to disentangle qibocal from qibolab.

stavros11 reviewed (commented):
Thank you for these updates. The platform page which I guess is the main purpose of this PR looks good to me, I mostly wrote some comments in the rest of the files. They are mostly simple so we could even complete quickly before merging this.

alecandido commented on doc/source/main-documentation/platform.rst:
```diff
@@ -3,201 +3,251 @@
+However, not all the configurations have to be channel configurations, e.g.
+:class:`OscillatorConfig` used for local oscillators.
+Moreover, since the configurations host actual parameters, it is pretty common that they
+are further specialized by instruments to add room for device-specific parameters.
```
It is definitely an option, but I wanted to keep this part instrument agnostic, avoiding specific references.

Maybe we could cook a mocked example. What do you think?

stavros11 commented on doc/source/main-documentation/circuits.rst:
```diff
@@ -0,0 +1,6 @@
+Circuits
+========
+
+Qibolab also provides a Qibo backend.
```
> Maybe we should also put some "work-in-progress" labels on some pages.

That would also be a good idea. For that page in particular we could just copy-paste my pretty minimal example since we already have this (or commit directly if it doesn't break any formatting) and add a "work-in-progress" to potentially revise in the future. I wouldn't do anything else in this PR.

alecandido replied:
Done in ae30703252924e0687d5e8bef045bc29c81ed1d2

stavros11 commented on doc/source/tutorials/instrument.rst:
```diff
@@ -131,3 +131,20 @@ Let's see a minimal example:
             # also the results are, in qibolab, specific objects that need some kind
```
> In any case, if this page is any relevant right now, we can certainly add Qblox. That's minimal effort.

Yes, the idea was just to add a `* Qblox` bullet under QM on top of this file (and the same for `main-documentation/drivers.rst`), since the effort is really minimal.

alecandido replied:
Done in 6642ce84b0e6af8b8a6704f40ca306b7b3c03b5b

alecandido committed:
docs: Drop deprecated qua lib, mention qblox

alecandido commented on doc/source/conf.py:
```diff
@@ -40,15 +34,13 @@
+    "icarusq_rfsoc_driver",
+    "keysight",
+    "qm",
+    "qualang_tools",
```
Done in 6642ce84b0e6af8b8a6704f40ca306b7b3c03b5b

alecandido committed:
docs: Replace sphinx-toolbox with sphinx-design

Because of transitive dependency issue, namely sphinx-prompt

https://github.com/sphinx-toolbox/sphinx-toolbox/issues/190

alecandido committed:
build: Update sphinx and furo versions

alecandido committed:
docs: Propagate dropdown switch to emulator page

alecandido committed:
docs: Add work-in-progress admonition on outdated pages
